CHANGELOG.md

```diff
@@ -1,5 +1,21 @@
 # Changelog
 
+## 14.3.28 (08/26/24)
```

Comment:
@camscale This should have the description of the security fix like other public security releases we published last week: https://github.com/gravitational/teleport/pull/45785/files.

Reply:
Done